pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/NonPersistentTopicStatsImpl.java

```diff
@@ -57,7 +62,11 @@
 
     @JsonProperty("publishers")
     public List<NonPersistentPublisherStats> getNonPersistentPublishers() {
-        return (List<NonPersistentPublisherStats>) nonPersistentPublishers;
+        return Stream.concat(nonPersistentPublishers.stream()
+                                .sorted(Comparator.comparing(NonPersistentPublisherStats::getProducerName)),
```

Comment:
sorting here may be a problem for users.
because users will tend to think that this list is always sorted by producerName, but there is documentation about this, and stuff like this tend to break in future refactors.

do you really need this list to be sorted ?

Reply:
> do you really need this list to be sorted ?

I don't strongly need the sorted order. It is for a consistent reason. Currently, publishers have some constant order even if they get stats in many times.
I'll remove the reordering operation if we don't have to support consistent orders.

Reply:
@eolivelli